### DISCUSSION TITLE
NEXT_PUBLIC_SERVER_URL value when promoting payloadcms site to Vercel?

### DISCUSSION THREAD

**Discussion by richleach** [vercel/community#5668] [Bug Report]
### Page to Investigate

https://thebeatentrail-supabase.vercel.app

### Steps to Reproduce

I developed a Payload site locally (localhost:3000) with a Supabase backend. All worked fine in dev.
Now I'm trying to host this site on Vercel and successfully ran the build process and have a site to see. Pages made from static files that were in my Github display fine. Problem is any page I made via the /admin comes up 404 and the error thrown in the console is

try {
let e = await fetch("".concat("https://thebeatentrail-supabase.vercel.app/", "/api/users/me"), {
method: "GET",
credentials: "include",
headers: {
"Content-Type": "application/json"
}
});
if (e.ok) {
let {user: t} = await e.json();
l(t || null),
s(t ? "loggedIn" : void 0)
} else
throw Error("An error occurred while fetching your account.")
} catch (e) {
throw l(null),
Error("An error occurred while fetching your account.")
}

Somewhere the application is trying to authorize "me" and makes an API call to "/api/users/me" but I think the url is not correct? In my environment variables section I assigned NEXT_PUBLIC_SERVER_URL = https://thebeatentrail-supabase.vercel.app/ , which is the dynamic url Vercel assigned to my site.... Can anyone confirm the value I should be using for NEXT_PUBLIC_SERVER_URL ?
What value(s) should I update and use in my .env vars when promoting my payloadcms site to Vercel? Didn't see anything in the docs....
Thanks, Rich

**Comment by richleach** [comment]
i've been trying to work through this issue as it is holding up my site from going live. In an effort to expedite this issue I was searching for some possible insight....
![image](https://github.com/vercel/community/assets/1168807/361b131c-cf44-4390-80ef-a9f4480c14e4)
...appears like a possible config issue/error might be afoot... The screenshot above indicates more than one CORS issue so I thought I'd add one or more of these errors to my whitelist (payload.config.ts)..... Trouble is, the fonts - as I understand Next.js builds - don't they download the fonts at build time so that you don't have to request them at page load time? And there's even a stripe.com call getting a CORS flag, and that's already in the cors: and csrf: sections.... so perhaps my updated question now is how would I deal with solving these CORS errors in Payload? I ask all that to get to the question of should I add "/api/users/me" to the cors: and csrf: sections so that Payload/Next.js let's it run?

**Reply by francois-pasquier** [reply]
I am facing the exact same issue, 404 on XHR calls and not knowing what to do with NEXT_PUBLIC_SERVER_URL as we should be using VERCEL_URL
